test/catalog/tablespace_handle_test.cpp

```diff
@@ -0,0 +1,44 @@
+#include <algorithm>
+#include <random>
+#include <vector>
+#include "catalog/catalog.h"
+#include "catalog/namespace_handle.h"
+#include "transaction/transaction_manager.h"
+#include "type/transient_value_peeker.h"
+#include "util/test_harness.h"
+#include "util/transaction_test_util.h"
+namespace terrier {
+
+struct TablespaceHandleTests : public TerrierTest {
+  void SetUp() override {
+    TerrierTest::SetUp();
+    txn_manager_ = new transaction::TransactionManager(&buffer_pool_, true, LOGGING_DISABLED);
+
+    txn_ = txn_manager_->BeginTransaction();
+    catalog_ = new catalog::Catalog(txn_manager_, txn_);
+  }
+
+  void TearDown() override {
+    txn_manager_->Commit(txn_, TestCallbacks::EmptyCallback, nullptr);
+
+    TerrierTest::TearDown();
+    delete catalog_;  // delete catalog first
+    delete txn_manager_;
+    delete txn_;
```

Comment:
Don't do this. Turn on the GC instead.

Reply:
This is part of the catalog PR we merged